torchvision/prototype/datapoints/_dataset_wrapper.py

```diff
@@ -39,8 +40,13 @@ def decorator(wrapper_factory):
 class VisionDatasetDatapointWrapper(Dataset):
     def __init__(self, dataset):
         dataset_cls = type(dataset)
-        wrapper_factory = WRAPPER_FACTORIES.get(dataset_cls)
-        if wrapper_factory is None:
+        for cls in itertools.takewhile(
+            lambda dataset_cls: dataset_cls is not datasets.VisionDataset, dataset_cls.mro()
+        ):
```

Comment:
can we just have a normal `for` + `if` checks please 🙏

`for-else` and `takewhile` are great when we understand what they do, but not everybody does

Reply:
(also we probably don't even need to check against VisionDataset? stopping at Object should be fine?)